File: src/main/java/net/sharplab/epubtranslator/core/service/EPubTranslatorService.java
package net.sharplab.epubtranslator.core.service;

import net.sharplab.epubtranslator.core.model.EPubFile;

import javax.enterprise.context.Dependent;

public interface EPubTranslatorService {

    EPubFile translate(EPubFile ePubFile, String srcLang, String dstLang);

}

File: src/main/java/net/sharplab/epubtranslator/core/service/EPubTranslatorService.kt
package net.sharplab.epubtranslator.core.service

import net.sharplab.epubtranslator.core.model.EPubFile

interface EPubTranslatorService {
    fun countCharacters(ePubFile: EPubFile): Map<String, Int>
    fun translate(ePubFile: EPubFile, srcLang: String, dstLang: String): Pair<EPubFile, TranslationFailure?>
}
